client/controllers/events/eventFiltration.coffee

```diff
@@ -23,6 +43,12 @@ Template.eventFiltration.onCreated ->
       filters.status =
         $in: status
 
+    selectedLocations = @selectedLocations.find().map (location) ->
+      location.name
+    if selectedLocations.length
+      locationProjection = "locations.#{@countryLevel.get()}"
+      filters[locationProjection] = $in: selectedLocations
```

Comment:
How does this look for a solution: f2179b4
I also noticed there's a case where a continent can be used. I suppose we should consider this?